### PR TITLE
Added unit tests for filters

### PR DESCRIPTION
**Pull request by iambibhas**

### CODE REVIEW

File: tests/test_filters.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        super(FilterTestCase, self).setUp()
+        self.now = datetime.utcnow()
+        self.test_user = TestUser()
+        self.test_avatar_size = ('100', '100')
+        self.test_avatar_url = u'//www.gravatar.com/avatar/test'
+
+    def tearDown(self):
+        super(FilterTestCase, self).tearDown()
+
+    def test_age(self):
```
Comment (shreyas-satish) [inline tests/test_filters.py]: Another case for minutes?

File: tests/test_filters.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+        ten_months = self.now - timedelta(days=10*30)
+        age = filters.age(ten_months)
+        self.assertEqual(age, u'10 months ago')
+
+        three_years = self.now - timedelta(days=3*12*30)
+        age = filters.age(three_years)
+        self.assertEqual(age, u'2 years ago')
+
+    def test_usessl(self):
```
Comment (shreyas-satish) [inline tests/test_filters.py]: Maybe another case for a url without the `http`?

File: tests/test_filters.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        with self.app.test_request_context('/'):
+            self.app.config['USE_SSL'] = False
+            ssled = filters.usessl('http://hasgeek.com')
+            self.assertEqual(ssled, 'http://hasgeek.com')
+
+            self.app.config['USE_SSL'] = True
+            ssled = filters.usessl('http://hasgeek.com')
+            self.assertEqual(ssled, 'https://hasgeek.com')
+
+    def test_nossl(self):
```
Comment (shreyas-satish) [inline tests/test_filters.py]: As above.

File: tests/test_filters.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        nossled = filters.nossl('https://hasgeek.com')
+        self.assertEqual(nossled, 'http://hasgeek.com')
+
+    def test_avatar_url(self):
+        avatar_url = filters.avatar_url(self.test_user)
+        self.assertEqual(avatar_url, u'//www.gravatar.com/avatar/00000000000000000000000000000000?d=mm')
+
+        self.test_user.set_avatar(self.test_avatar_url)
```
Comment (shreyas-satish) [inline tests/test_filters.py]: Maybe another case for a test user without an email?

Comment (iambibhas) [inline tests/test_filters.py]: @shreyas-satish the case before this one, is for a user who has neither a prefilled avatar, nor an email. hence that `0000..` gravatar url.

File: tests/fixtures.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+class BaseframeTestCase(unittest.TestCase):
+    def setUp(self):
+        self.app = Flask(__name__)
+        baseframe.init_app(self.app, requires=['baseframe'])
+
+    def tearDown(self):
+        pass
+
+
+class TestUser:
```
Comment (jace) [inline tests/fixtures.py]: Always subclass `object`

File: tests/test_filters.py [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+class FilterTestCase(BaseframeTestCase):
+    def setUp(self):
+        super(FilterTestCase, self).setUp()
+        self.now = datetime.utcnow()
+        self.test_user = TestUser()
+        self.test_avatar_size = ('100', '100')
+        self.test_avatar_url = u'//www.gravatar.com/avatar/test'
+
+    def tearDown(self):
+        super(FilterTestCase, self).tearDown()
```
Comment (jace) [inline tests/test_filters.py]: Is this method necessary at all?

File: tests/test_filters.py [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+        self.test_user.set_avatar(self.test_avatar_url)
+        avatar_url = filters.avatar_url(self.test_user, self.test_avatar_size)
+        self.assertEqual(avatar_url, self.test_avatar_url + '?size=' + u'x'.join(self.test_avatar_size))
+
+        self.test_user.set_avatar(None)
+        self.test_user.set_email('foobar@foo.com')
+        avatar_url = filters.avatar_url(self.test_user, self.test_avatar_size)
+        hash = md5sum(self.test_user.email)
+        self.assertEqual(avatar_url, u'//www.gravatar.com/avatar/' + hash + u'?d=mm&s=' + u'x'.join(self.test_avatar_size))
```
Comment (jace) [inline tests/test_filters.py]: Doesn't this filter also return an avatar URL if it's explicitly set on the user object? Test that.

File: tests/test_filters.py (hunk 4 above)
Comment (iambibhas) [inline tests/test_filters.py]: @jace this case is where the user object already has an avatar set.

File: tests/test_filters.py (hunk 6 above)
Comment (iambibhas) [inline tests/test_filters.py]: nope. just thought I'll keep it for uniformity's sake. i'll remove these.

**Pull request merged**